packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts

```diff
@@ -15,21 +15,56 @@
 // *****************************************************************************
 
 import { injectable, inject } from '@theia/core/shared/inversify';
-import { CommandRegistry, CommandContribution } from '@theia/core/lib/common';
+import { CommandRegistry, CommandContribution, Command } from '@theia/core/lib/common';
 import { OpenUriCommandHandler } from './commands';
 import URI from '@theia/core/lib/common/uri';
+import { TreeViewWidget } from './view/tree-view-widget';
+import { CompositeTreeNode, Widget, codicon } from '@theia/core/lib/browser';
+import { TabBarToolbarContribution, TabBarToolbarRegistry } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+import { PluginViewWidget } from './view/plugin-view-widget';
 
 @injectable()
-export class PluginApiFrontendContribution implements CommandContribution {
+export class PluginApiFrontendContribution implements CommandContribution, TabBarToolbarContribution {
 
     @inject(OpenUriCommandHandler)
     protected readonly openUriCommandHandler: OpenUriCommandHandler;
 
+    static readonly COLLAPSE_ALL_COMMAND = Command.toDefaultLocalizedCommand({
+        id: 'treeviews.collapseAll',
+        iconClass: codicon('collapse-all')
+    });
+
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(OpenUriCommandHandler.COMMAND_METADATA, {
             execute: (arg: URI) => this.openUriCommandHandler.execute(arg),
             isVisible: () => false
         });
+        commands.registerCommand(PluginApiFrontendContribution.COLLAPSE_ALL_COMMAND, {
+            execute: (widget: Widget) => {
+                if (widget instanceof PluginViewWidget && widget.widgets[0] instanceof TreeViewWidget) {
+                    const model = widget.widgets[0].model;
```

Comment:
Do you think it might be valuable to have that 'collapseTree' functionality generically in the model or is it specific to the tree that is usually provided through the plugins? To me, it looks quite generic but I know that roots might be handled differently depending on the expected tree structure.

Reply:
Yes, this sounds quite generic, but I don't see that refactoring as a necessary part of this PR.

Reply:
Understood.